purchases/src/main/kotlin/com/revenuecat/purchases/virtualcurrencies/VirtualCurrencyManager.kt

```diff
@@ -85,16 +100,46 @@ internal class VirtualCurrencyManager(
             return null
         }
 
-        return deviceCache.getCachedVirtualCurrencies(appUserID = appUserID)
+        val cachedVirtualCurrencies: VirtualCurrencies? = deviceCache.getCachedVirtualCurrencies(appUserID = appUserID)
+        if (cachedVirtualCurrencies == null) {
+            log(LogIntent.DEBUG) {
+                VirtualCurrencyStrings.NO_CACHED_VIRTUAL_CURRENCIES
+            }
+        }
+
+        return cachedVirtualCurrencies
     }
 
-    @Suppress("UnusedParameter", "ForbiddenComment")
     private fun fetchVirtualCurrenciesFromBackend(
         appUserID: String,
         isAppBackgrounded: Boolean,
         callback: GetVirtualCurrenciesCallback,
-    ): VirtualCurrencies {
-        // TODO: Implement this
-        return VirtualCurrencies(all = emptyMap())
+    ) {
+        backend.getVirtualCurrencies(
+            appUserID = appUserID,
+            appInBackground = isAppBackgrounded,
+            onSuccess = { callback.onReceived(it) },
+            onError = { callback.onError(it) },
+        )
+    }
+
+    private fun fetchedVirtualCurrenciesFromBackendCallback(
```

Comment:
Hmm wdyt of something like `handleVirtualCurrenciesRequestResult`?

Then, we could rename `originalCallback` to `completion`.